tests/unit/test_resolver.py

```diff
@@ -93,3 +97,71 @@ def fetch(req: Request) -> RequestOutput:
         assert resolver.resolve_packages(request) == COMBINED_OUTPUT
 
     assert calls_by_pkgtype == ["gomod", "npm", "pip"]
+
+
+@pytest.mark.parametrize(
+    "flags",
+    [
+        pytest.param(["dev-mode"], id="dev-mode-true"),
+        pytest.param([], id="dev-mode-false"),
+    ],
+)
+def test_dev_mode(flags: list[str]) -> None:
+    with mock.patch.dict(
+        resolver._dev_package_managers,
+        values={"shrubbery": lambda _: RequestOutput.empty()},
+        clear=True,
+    ):
+
+        class foo:
+            type: str
+
+        dev_package_input = foo()
+        dev_package_input.type = "shrubbery"
+
+        request = mock.Mock()
+        request.flags = flags
+        request.packages = [dev_package_input]
+
+        if flags:
+            assert resolver.resolve_packages(request) == RequestOutput(
+                components=[], build_config=BuildConfig(environment_variables=[], project_files=[])
+            )
+        else:
+            expected_error = re.escape("Package manager(s) not yet supported: shrubbery")
+            try:
+                resolver.resolve_packages(request)
+            except UnsupportedFeature as exc:
+                assert re.match(expected_error, str(exc))
```

Comment:
Can we rewrite this piece using `pytest.raises`? I think it makes for a more readable syntax.

Reply:
It fails for me here. Pytest says the exception isn't being raised.

Reply:
`FAILED tests/unit/test_resolver.py::test_dev_mode[dev-mode-false] - Failed: DID NOT RAISE <class 'cachi2.core.errors.UnsupportedFeature'>`

Reply:
And that's with a completely clean repo (except for my code changes), with make and tox generating everything from scratch :shrug: